src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/ScalarWaveGr.hpp

```diff
@@ -53,13 +60,21 @@ class ScalarWaveGr : public MarkAsAnalyticData {
       ScalarFieldData::volume_dim == BackgroundGrData::volume_dim,
       "Scalar field data and background spacetime data should have the same "
       "spatial dimensionality. Currently provided template arguments do not.");
+  static_assert(tmpl::list_contains_v<typename ScalarFieldData::tags,
+                                      ScalarWave::Tags::Psi> or
+                    tmpl::list_contains_v<typename ScalarFieldData::tags,
+                                          CurvedScalarWave::Tags::Psi>,
+                "The scalar field data needs to be able to return either the "
+                "ScalarWave or the CurvedScalarWave evolved variables");
 
  public:
   static constexpr size_t volume_dim = ScalarFieldData::volume_dim;
-
+  static constexpr bool is_curved =
+      tmpl::list_contains_v<typename ScalarFieldData::tags,
+                            CurvedScalarWave::Tags::Psi>;
   struct ScalarField {
     using type = ScalarFieldData;
-    static constexpr Options::String help = {"Flat space scalar field."};
+    static constexpr Options::String help = {"Scalar field."};
```

Comment:
[optional] Maybe expand to `The scalar field system data.` to make it clear it sets the subsystem?